external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py

```diff
@@ -300,6 +300,15 @@ def _init_model_cfg(self):
         else:
             cfg_path = os.path.join(base_dir, 'model.py')
         cfg = MPAConfig.fromfile(cfg_path)
+
+        # To initialize different HP for multi-label classification / Support HP change via CLI & UI
+        if self._multilabel:
+            template = MPAConfig.fromfile(self.template_file_path)
+            template_params = template.hyper_parameters.parameter_overrides.learning_parameters
+            incoming_params = self._hyperparams.learning_parameters
+            if cfg.get('runner', False) and (template_params.num_iters.default_value != incoming_params.num_iters):
+                cfg.runner.max_epochs = incoming_params.num_iters
```

Comment:
Do we update # epochs only?

Reply:
Yes, another configs (lr, warmup_iters) are discarded by performance drop on class-incremental learning.

Reply:
Actually, it's hard for me to understand this code. If it is different from the default value, change max_epochs, and in the case of the default value, do you mean using the epoch of the recipe? 

In this case, isn't it possible to actually train with a different value from the value shown in the template?

Reply:
No, 
If incoming parameter (with CLI, that might be `params --learning_parameters.num_iters`) is **same** with default value of template, replace max epoch from 20 -> 90
But if incoming parameter is **different** with default value of template, **max_epoch should be changed into incoming parameters**. So, this code does not change max epoch to 90 to support control num_iters via OTE-CLI or SC UI.

Reply:
Seems like no other choice currently. Since we did not separate the template of multi-class and multi-label(which requires some huge code change), we should have shared the parameters even if it has a best results at the different configuration. According to @JihwanEom , that features(early-stop and more epochs) influences hugely in performance for multi-label, while they are not in multi-class (I also tried to match to those parameters what he founds. But ended up needing additional hp search for early-stop/epochs that fits for multi-class classification. That was why I shared the early-stop exp. suddenly today.)

Reply:
This would be considered a bug in the eyes of the user.  Also, I actually felt that early-stop was better after looking at the experimental results.  Of course, it would be nice to proceed when there are more experimental results, but this feels like a bug to me.

@goodsong81 What do you think?

Reply:
> I think the epochs will be 90 in training, but the value displayed in the SC UI will be 20 so wouldn't this be a problem?

Yeap sure, that's an limitation of this PR. But multi-label classification performance significantly drops without change of max_epoch up to 40%. As vlad said, I believe this is the limitation of SC design. I expect this issue will be resolved on next version of OTE/SC.

Reply:
I also think it look like a bug. But changing default setup to early-stop and larger epoch is quite not good choice for multi-class classification hastily without additional exp. In one template with cifar10, early stop was helpful. But generally, I think it's not good idea to use it in order to earn 1% accuracy taking 2x/3x more epochs.(some are even better both in accuracy and time with no_early_stop setting) It needs to run more experiments definitely with different lr or other param, which takes long period to search and analyze the performance of it.

Reply:
I agree with the soobee.  But I don't think it's appropriate to leave it as it is, and some other way to make it not look like a bug should be added in this PR..

Reply:
Actually, I don't even have any ideas also..